ooiservices/app/main/oauth.py

```diff
@@ -0,0 +1,85 @@
+from rauth import OAuth2Service
+from flask import current_app, request, redirect
+import json
+
+
+class OAuthSignIn(object):
+    providers = None
+
+    def __init__(self, provider_name):
+        self.provider_name = provider_name
+        self.consumer_id = current_app.config['CILOGON_ID']
+        self.consumer_secret = current_app.config['CILOGON_SECRECT']
+
+    def authorize(self):
+        pass
+
+    def callback(self):
+        pass
+
+    def get_callback_url(self):
+        return current_app.config['UI_URL'] + '/callback/cilogon'
+
+    @classmethod
+    def get_provider(self, provider_name):
+        if self.providers is None:
+            self.providers = {}
+            for provider_class in self.__subclasses__():
+                provider = provider_class()
+                self.providers[provider.provider_name] = provider
+        return self.providers[provider_name]
+
+
+class CILogonSignIn(OAuthSignIn):
+    def __init__(self):
+        super(CILogonSignIn, self).__init__('cilogon')
+        self.service = OAuth2Service(
+            name='cilogon',
+            client_id=self.consumer_id,
+            client_secret=self.consumer_secret,
+            authorize_url='https://test.cilogon.org/authorize',
```

Comment:
Let's move these to the config file as well, knowing that they will change early next year when cilogon is formally released.